sycl/test-e2e/CMakeLists.txt

```diff
@@ -69,6 +69,13 @@ endif()
 
 set(SYCL_E2E_CLANG_CXX_FLAGS "${SYCL_E2E_CLANG_CXX_FLAGS} -Werror")
 
+find_package(CUDAToolkit)
+
+if(CUDAToolkit_FOUND)
+  set(CUDA_LIBS_DIR "${CUDAToolkit_LIBRARY_DIR}")
```

Comment:
what overrides what if both this and explicit params are present?

Reply:
Typically, explicit command line arguments override defaults.

What error do we get if CMake can't find CUDA toolkit and user hasn't set the paths via cmd options? Does it make sense to add diagnostics to CMake files or LIT infrastructure already provides good diagnostics?

Reply:
right now we get no error, we just dont enable the `cuda_dev_kit` feature.

Reply:
Doesn't this code override CMake `CUDA_INCLUDE`/`CUDA_LIBS_DIR` variables passed via command line?

Reply:
ah i didnt consider anyone using cmake command line, ill fix it to not do the autodetection if theyre set